src/transactions/ManageBuyOfferOpFrame.cpp

```diff
@@ -0,0 +1,198 @@
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "transactions/ManageBuyOfferOpFrame.h"
+#include "ledger/LedgerTxn.h"
+#include "ledger/LedgerTxnEntry.h"
+#include "ledger/LedgerTxnHeader.h"
+#include "ledger/TrustLineWrapper.h"
+#include "transactions/OfferExchange.h"
+#include "transactions/TransactionUtils.h"
+
+namespace stellar
+{
+
+ManageBuyOfferOpFrame::ManageBuyOfferOpFrame(Operation const& op,
+                                             OperationResult& res,
+                                             TransactionFrame& parentTx)
+    : BaseManageOfferOpFrame(
+          op, res, parentTx, op.body.manageBuyOfferOp().selling,
+          op.body.manageBuyOfferOp().buying, op.body.manageBuyOfferOp().offerID,
+          op.body.manageBuyOfferOp().price, false)
+    , mManageBuyOffer(mOperation.body.manageBuyOfferOp())
+{
+}
+
+bool
+ManageBuyOfferOpFrame::isVersionSupported(uint32_t protocolVersion) const
+{
+    return protocolVersion >= 11;
+}
+
+bool
+ManageBuyOfferOpFrame::isDeleteOffer()
+{
+    return mManageBuyOffer.buyAmount == 0;
+}
+
+int64_t
+ManageBuyOfferOpFrame::getOfferBuyingLiabilities()
+{
+    auto res = exchangeV10WithoutPriceErrorThresholds(
+        mManageBuyOffer.price, INT64_MAX, INT64_MAX, INT64_MAX,
```

Comment:
didn't realize that we're using this price and not the inverse (and that's what the cap says)

Reply:
More that it doesn't seem super natural to me as offers are expressed in terms of selling things, so buying implies using the semantics for the other half of the book maybe?

If I have an offer on the book:
* sell 1000 JPY for USD at price 0.009 (manageSellOffer on the "sell JPY for USD" book)

That I try to take entirely (1 is current CAP006, 2 is using the inverse):
 1. buy 1000 JPY for USD at price 110 (manageBuyOffer on the "sell USD for JPY" book)
 2. buy 1000 JPY for USD at price 0.009 (manageBuyOffer on the "sell USD for JPY" book)

Right now we only "sell" (the only thing we can do now)
sell 9 USD for JPY at price 110 (manageSellOffer on the "sell USD for JPY" book)

Maybe it's a UI thing. I don't know.

Reply:
@jonjove @MonsieurNicolas chipping in here.. this is what I’d expect as a consumer of these APIs:

when dealing with the XLM/USD market:
sell 100 XLM at $0.095
buy 100 XLM at $0.090

converting the same actions above to USD/XLM representation:
sell 9.5 USD at price 11.1111111
buy 9.0 USD at price 10.5263158

i.e. prices (buy and sell) should always be in the quote (counter) currency where for a market X/Y, X is the base currency and Y is the quote currency.
Also, amounts should always be specified in the base currency as in the examples above.